src/main/java/org/folio/rest/impl/VoucherAcquisitionsUnitAssignmentsHelper.java

```diff
@@ -0,0 +1,64 @@
+package org.folio.rest.impl;
+
+import static org.folio.invoices.utils.HelperUtils.buildQuery;
+import static org.folio.invoices.utils.HelperUtils.handleDeleteRequest;
+import static org.folio.invoices.utils.HelperUtils.handleGetRequest;
+import static org.folio.invoices.utils.HelperUtils.handlePutRequest;
+import static org.folio.invoices.utils.ResourcePathResolver.*;
+
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import org.folio.rest.jaxrs.model.AcquisitionsUnitAssignment;
+import org.folio.rest.jaxrs.model.AcquisitionsUnitAssignmentCollection;
+
+import io.vertx.core.Context;
+import io.vertx.core.json.JsonObject;
+import me.escoffier.vertx.completablefuture.VertxCompletableFuture;
+
+public class VoucherAcquisitionsUnitAssignmentsHelper extends AbstractHelper {
+
+  private static final String GET_VOUCHER_UNIT_ASSIGNMENTS_BY_QUERY = resourcesPath(VOUCHER_ACQUISITIONS_UNIT_ASSIGNMENTS) + SEARCH_PARAMS;
+
+  VoucherAcquisitionsUnitAssignmentsHelper(Map<String, String> okapiHeaders, Context ctx, String lang) {
+    super(okapiHeaders, ctx, lang);
+  }
+
+  CompletableFuture<AcquisitionsUnitAssignmentCollection> getAcquisitionsUnitAssignments(String query, int offset, int limit) {
+    CompletableFuture<AcquisitionsUnitAssignmentCollection> future = new VertxCompletableFuture<>(ctx);
+
+    try {
+      String endpoint = String.format(GET_VOUCHER_UNIT_ASSIGNMENTS_BY_QUERY, limit, offset, buildQuery(query, logger), lang);
+
+      handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
+        .thenApply(jsonUnits -> jsonUnits.mapTo(AcquisitionsUnitAssignmentCollection.class))
```

Comment:
From my point of view mapping of the collection potentially could be a blocking operation. You could run this mapping on a separate thread worker something like this:
```
handleGetRequest(endpoint, httpClient, ctx, okapiHeaders, logger)
        .thenCompose(jsonUnits -> VertxCompletableFuture.supplyBlockingAsync(ctx, () -> {
            if (logger.isInfoEnabled()) {
              logger.info("Successfully retrieved assignments list", jsonUnits.encodePrettily());
            }
            return jsonUnits.mapTo(AcquisitionsUnitAssignmentCollection.class);
          })
        )
        .thenAccept(future::complete)
        .exceptionally(t -> {
          future.completeExceptionally(t.getCause());
          return null;
        });

```

Reply:
Makes sense to map this on a separate thread if collection size grows larger. For now, I have added it. Nice point!